### PR TITLE
Tidy up module.

### PR DESCRIPTION
**Pull request by ben-eb**
* Switch from tape to ava.
* Switch from jshint to eslint.
* Update CI configuration.
* Use ES2015.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,5 +41,8 @@
     "alphanum-sort": "^1.0.1",
     "postcss": "^5.0.4",
     "uniqs": "^2.0.0"
+  },
+  "ava": {
+    "require": "babel-core/register"
```
Comment (TrySound) [inline package.json]: Why this?

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,74 @@
+root: true
```
Comment (TrySound) [inline .eslintrc]: I think it's only editorconfig feature.

File: .eslintrc [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    objectLiteralShorthandMethods: true
+    objectLiteralShorthandProperties: true
+    octalLiterals: true
+    regexUFlag: true
+    regexYFlag: true
+    restParams: true
+    spread: true
+    superInFunctions: true
+    templateStrings: true
+    modules: true
```
Comment (TrySound) [inline .eslintrc]: modules twice

File: package.json (hunk 1 above)
Comment (ben-eb) [inline package.json]: That's so we don't have to transpile the tests/source before running the tests.

Comment (TrySound) [inline package.json]: Ava do it out of the box. :)

File: .eslintrc (hunk 2 above)
Comment (ben-eb) [inline .eslintrc]: It's eslint too, :smile: 

http://eslint.org/docs/user-guide/configuring

File: .eslintrc (hunk 3 above)
Comment (ben-eb) [inline .eslintrc]: Ah, whoops! Thanks. :+1:

File: package.json (hunk 1 above)
Comment (ben-eb) [inline package.json]: Without it an exception is thrown regarding import statements. :smile:

File: .eslintrc (hunk 2 above)
Comment (TrySound) [inline .eslintrc]: Okay :)

**Pull request merged**